File: clients/php/src/TableClient.php
<?php

namespace Zeeplabs\Orbit;

class TableClient
{
    public function __construct(
        private readonly OrbitClient $client,
        private readonly string $table
    ) {}

    public function findMany(
        ?int $limit = null,
        ?int $offset = null,
        ?string $order = null,
        ?array $filters = null
    ): array {
        $params = [];
        if ($limit !== null) $params['limit'] = (string) $limit;
        if ($offset !== null) $params['offset'] = (string) $offset;
        if ($order !== null) $params['order'] = $order;
        if ($filters !== null) $params = array_merge($params, $filters);

        $qs = http_build_query($params);
        $path = $this->table . ($qs ? '/?' . $qs : '');
        return $this->client->request('GET', $path);
    }

    public function findById(string $id): array
    {
        return $this->client->request('GET', "{$this->table}/{$id}");
    }

    public function create(array $data): array
    {
        return $this->client->request('POST', "{$this->table}/", $data);
    }

    public function update(string $id, array $data): array
    {
        return $this->client->request('PATCH', "{$this->table}/{$id}", $data);
    }

    public function replace(string $id, array $data): array
    {
        return $this->client->request('PUT', "{$this->table}/{$id}", $data);
    }

    public function delete(string $id): void
    {
        $this->client->request('DELETE', "{$this->table}/{$id}");
    }
}

File: clients/java/src/main/java/com/zeeplabs/orbit/TableClient.java
package com.zeeplabs.orbit;

import com.zeeplabs.orbit.Types.ListResponse;
import java.util.HashMap;
import java.util.Map;

public class TableClient {

    private final OrbitClient client;
    private final String table;

    TableClient(OrbitClient client, String table) {
        this.client = client;
        this.table = table;
    }

    public ListResponse findMany(Integer limit, Integer offset, String order, Map<String, String> filters) {
        Map<String, String> params = new HashMap<>();
        if (limit != null) params.put("limit", String.valueOf(limit));
        if (offset != null) params.put("offset", String.valueOf(offset));
        if (order != null) params.put("order", order);
        if (filters != null) params.putAll(filters);

        String qs = client.buildQuery(params);
        return client.request("GET", table + "/" + qs, null, ListResponse.class);
    }

    public Map<String, Object> findById(String id) {
        return client.request("GET", table + "/" + id, null, Map.class);
    }

    public Map<String, Object> create(Map<String, Object> data) {
        return client.request("POST", table + "/", data, Map.class);
    }

    public Map<String, Object> update(String id, Map<String, Object> data) {
        return client.request("PATCH", table + "/" + id, data, Map.class);
    }

    public Map<String, Object> replace(String id, Map<String, Object> data) {
        return client.request("PUT", table + "/" + id, data, Map.class);
    }

    public void delete(String id) {
        client.request("DELETE", table + "/" + id, null, null);
    }
}
